examples/custom-resources/oidc/oidc.yaml

```diff
@@ -9,5 +9,6 @@ spec:
     authEndpoint: https://keycloak.example.com/realms/master/protocol/openid-connect/auth
     tokenEndpoint: http://keycloak.default.svc.cluster.local:8080/realms/master/protocol/openid-connect/token
     jwksURI: http://keycloak.default.svc.cluster.local:8080/realms/master/protocol/openid-connect/certs
+    endSessionEndpoint: https://keycloak.example.com/realms/master/protocol/openid-connect/logout
```

Comment:
Should we make these endpoints consistent?
Pick http://keycloak.default.svc.cluster.local:8080 or https://keycloak.example.com

Reply:
I think this was because the IDP is deployed at the same cluster as NIC and the domain `example.com` is not real. As you can see, `https://keycloak.example.com/realms/master/protocol/openid-connect/auth` is the URL NGINX would redirect the user browser to login, but `tokenEndpoint` and ` jwksURI` are URL that NGINX itself would send requests to as the OIDC client. if we want to make the endpoints more consistent, it will have to be `https://keycloak.example.com/`, but I'll also need to investigate if or how NGINX can resolve `keycloak.example.com` with kubedns in the cluster as its resolver.

Reply:
ok, thanks for clarifying @haywoodsh.  Let's leave it as is for this PR and open an issue to follow on.

Reply:
FYI I got this error log when I set `tokenEndpoint: https://keycloak.example.com/realms/master/protocol/openid-connect/token`  but the login and logout action seem to be fine, and NGINX seems to be able to resolve the domain name too
```
2024/08/06 16:06:36 [info] 79#79: *115 recv() failed (104: Connection reset by peer) while sending to client, client: 127.0.0.1, server: keycloak.example.com, request: "POST /realms/master/protocol/openid-connect/token HTTP/1.0", upstream: "http://10.42.0.51:8080/realms/master/protocol/openid-connect/token", host: "keycloak.example.com"
```